cfpb/consumerfinance.gov
# Update esbuild and refactor to support context API

## Changes

- Update to latest esbuild
-  Refactor build script to support context API introduced in esbuild 0.17.0 so that watch command works.


## How to test this PR

1. `yarn build` should build the assets.
2. `yarn build scripts` should build only the JS scripts.
3. `yarn build styles` should build only the CSS.
4. `yarn watch` should watch the JS and the CSS (make a change to the JS or to the Less files and see that the changed file is logged in the terminal). Ctrl+c to end watch.

## Notes and todos

- Changing from build to rebuild we don't get an output summary any longer. The build still works however. I opened https://github.com/evanw/esbuild/issues/2886 for guidance.
- The watch command is only watching scripts and styles. In the future, we might want to expand that to all files that are part of the build.
